docs: update docs about prodMode.

JeanMeche opened the pull request:
Following #49723, this commit updates the doc to reflect the actual message prompted in the console.

alan-agius4 reviewed (approved):
LGTM

alan-agius4 commented:
Caretaker: the CI failure is unrelated.

AndrewKushnir commented on aio/content/guide/deployment.md:
```diff
@@ -257,15 +257,14 @@ See [`ng build`](cli/build) for more about CLI build options and what they do.
-### Enable runtime production mode
+### Production mode
 
 In addition to build optimizations, Angular also has a runtime production mode.
```
I'd propose replacing the whole content of this section with something like this:
```
When you run an application locally using `ng serve`, Angular uses the development mode configuration
at runtime. The development mode at runtime enables extra safety checks, more detailed error messages
and debugging utilities, such as the expression-changed-after-checked detection. Angular outputs
a message in a browser console to indicate that the development mode is enabled.

Those extra checks are helpful during the development, but they require an extra code in a bundle, which is
undesirable in production. To ensue that there are no implications on the bundle size, the build optimizer
removes the development-only code from the bundle when building in production mode.
```

I think it'd be great to avoid mentioning a specific message content, so that we don't get in a situation when it's out of sync with the code (like it happened after landing https://github.com/angular/angular/pull/49723).

JeanMeche replied:
Thanks, It's nicely written, I used most of it. Let me know what you think of the few additions I made !

AndrewKushnir reviewed (commented):
@JeanMeche thanks for catching this! 👍 

I've left a comment to update the section and avoid mentioning a specific content, so we don't get out of sync in the future. Please let me know what you think.

AndrewKushnir commented on aio/content/guide/deployment.md:
```diff
@@ -247,30 +247,28 @@ The `production` configuration engages the following build optimization features
 |:---                                                   |:---     |
 | [Ahead-of-Time (AOT) Compilation](guide/aot-compiler) | Pre-compiles Angular component templates.                                |
-| [Production mode](#enable-prod-mode)                  | Deploys the production environment which enables *production mode*.      |
+| [Production mode](#prod-mode)                         | Disables safety checks and debugging utilities |
```
I think we can rephrase this a bit to make more positive :) 
```suggestion
| [Production mode](#prod-mode)                         | Optimizes the application for the best runtime performance |
```

JeanMeche replied:
I like your point of view 👍

AndrewKushnir reviewed (commented):
Looks great, just 1 small comment.

JeanMeche committed:
docs: update docs about prodMode.

Following angular#49723, this commit updates the doc to reflect the actual message prompted in the console.

AndrewKushnir commented on aio/content/guide/deployment.md:
```diff
@@ -247,30 +247,28 @@ The `production` configuration engages the following build optimization features
 
-*Production mode* improves application performance by disabling development-only safety checks and debugging utilities, such as the expression-changed-after-checked detection.
-Building your application with the production configuration automatically enables Angular's runtime production mode.
+When using the CLI, Production mode is enabled by the `optimization` option.
```
I'd propose keeping an existing sentence here:
```suggestion
Building your application with the production configuration automatically enables Angular's runtime production mode.
```
With the current sentence it sounds more like it should be configured (when it fact it automatically enabled/disabled based on the build mode).

AndrewKushnir reviewed (approved):
Looks great, thanks for addressing the feedback! 👍

AndrewKushnir commented:
This PR was merged into the repository by commit 893eef39fca23be71b92c9cf4653aab78fcb8de4.
